ui/src/app/components/apps-panel/runtime-state-button.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from 'react';
+import { useEffect, useState } from 'react';
+import { faPause, faPlay } from '@fortawesome/free-solid-svg-icons';
+import { faSyncAlt } from '@fortawesome/free-solid-svg-icons/faSyncAlt';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+import { RuntimeStatus, Workspace } from 'generated/fetch';
+
+import { Clickable } from 'app/components/buttons';
+import { FlexColumn } from 'app/components/flex';
+import { cond, switchCase } from 'app/utils';
+import {
+  RuntimeStatusRequest,
+  useRuntimeStatus,
+} from 'app/utils/runtime-utils';
+
+import { buttonStyles } from './utils';
+
+export const RuntimeStateButton = (props: { workspace: Workspace }) => {
+  const {
+    workspace: { namespace, googleProject },
+  } = props;
+
+  const [status, setRuntimeStatus] = useRuntimeStatus(namespace, googleProject);
+
+  // transition states, so we don't have to wait for polling
+  const [pausing, setPausing] = useState(false);
+  const [resuming, setResuming] = useState(false);
+
+  // when polling updates the status value, clear our transition states
+  useEffect(() => {
+    if (pausing) {
+      setPausing(false);
+    }
+    if (resuming) {
+      setResuming(false);
```

Comment:
Should pausing be updated if resuming's state is updated?

Reply:
The intention here is to avoid re-renders.  Only clear those values which are set.